cli/azd/pkg/devcenter/environment_store.go

```diff
@@ -0,0 +1,212 @@
+package devcenter
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/contracts"
+	"github.com/azure/azure-dev/cli/azd/pkg/devcentersdk"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment"
+)
+
+const (
+	ConfigPath                                 = "platform.config"
+	RemoteKindDevCenter environment.RemoteKind = "devcenter"
+)
+
+// EnvironmentStore is a remote environment data store for devcenter environments
+type EnvironmentStore struct {
+	config          *Config
+	devCenterClient devcentersdk.DevCenterClient
+	prompter        *Prompter
+	manager         Manager
+}
+
+// NewEnvironmentStore creates a new devcenter environment store
+func NewEnvironmentStore(
+	config *Config,
+	devCenterClient devcentersdk.DevCenterClient,
+	prompter *Prompter,
+	manager Manager,
+) environment.RemoteDataStore {
+	return &EnvironmentStore{
+		config:          config,
+		devCenterClient: devCenterClient,
+		prompter:        prompter,
+		manager:         manager,
+	}
+}
+
+// EnvPath returns the path for the environment
+func (s *EnvironmentStore) EnvPath(env *environment.Environment) string {
+	return fmt.Sprintf("projects/%s/users/me/environments/%s", s.config.Project, env.GetEnvName())
+}
```

Comment:
Do we need to pass `env` ?  It looks like all we need is the `env-name`

If we change it to `string`, we can no longer worry about `env == nil` and a null-ref here

Reply:
This is function is part of the `environment.DataStore` interface. It performs the action against the specified environment.  In this implementation it only needs the name, but another implementation could require some other member from the env instance.